pkg/controller/daemon/util/daemonset_util.go

```diff
@@ -225,3 +227,43 @@ func AppendNoScheduleTolerationIfNotExist(tolerations []v1.Toleration) []v1.Tole
 
 	return tolerations
 }
+
+// GetTargetNodeName get the target node name of DaemonSet pods. If ScheduleDaemonSetPods enabled, it will also
+// retrieve node name of pending pods from NodeAffinity; otherwise, `.spec.NodeName` will be return.
```

Comment:
The comment isn't quite correct. Non-nil `.spec.NodeName` will be returned. Otherwise, we retrieve it from NodeAffinity if ScheduleDaemonSetPods enabled -- which means that as long as `.spec.NodeName` is set we don't check if the feature is enabled or not.